pkg/api/routes.go

```diff
@@ -55,11 +58,36 @@ func NewRouteHandler(c *Controller) *RouteHandler {
 }
 
 func (rh *RouteHandler) SetupRoutes() {
+	// first get Auth middleware in order to first setup openid/ldap/htpasswd, before oidc provider routes are setup
+	authHandler := AuthHandler(rh.c)
+
+	applyCORSHeaders := getCORSHeadersHandler(rh.c.Config.HTTP.AllowOrigin)
+
+	if isOpenIDAuthEnabled(rh.c.Config) {
+		// login path for openID
+		rh.c.Router.HandleFunc(constants.LoginPath, rh.AuthURLHandler())
+
+		// logout path for openID
+		rh.c.Router.HandleFunc(constants.LogoutPath, applyCORSHeaders(rh.Logout)).
+			Methods(zcommon.AllowedMethods("POST")...)
+
+		// callback path for openID
+		for provider, relyingParty := range rh.c.RelyingParties {
+			if IsOauth2Supported(provider) {
+				rh.c.Router.HandleFunc(constants.CallbackBasePath+fmt.Sprintf("/%s", provider),
```

Comment:
If network policy doesn't allow inbound connections, this callback wont work? Call this out in the commit msg and docs

Reply:
Add a docs item for social login for the feature, caveats and notes.